File: algorithms/cpp/171.cpp
#include <iostream>
using namespace std;

class Solution
{
public:
    int titleToNumber(string s)
    {
        int ans = 0;
        for ( unsigned int i = 0; i < s.size(); i++ )
            ans = ans*26+s[i]-'A'+1;
        return ans;
    }
};

int main()
{
    Solution solution;
    cout << solution.titleToNumber("AB") << endl;
    return 0;
}

File: algorithms/python/171.py
class Solution(object):
    def titleToNumber(self, s):
        """
        :type s: str
        :rtype: int
        """
        s, base = s[::-1], 1
        rls = 0
        for ch in s:
            rls += (ord(ch)-ord('A')+1) * base
            base *= 26
        return rls
